src/main/java/fr/moribus/imageonmap/image/MapInitEvent.java

```diff
@@ -131,4 +146,115 @@ public void onPlayerInventoryPlace(InventoryClickEvent event) {
 
         }
     }
+
+    @EventHandler
+    public void onEntityDamageByEntityEvent(EntityDamageByEntityEvent event) {
+        //Negate entity interaction with item frame containing IoM maps.
+
+        Entity entity = event.getEntity();
+        if (!(entity instanceof ItemFrame)) {
+            return;
+        }
+        Entity damager = event.getDamager();
+        if (damager instanceof Player) {
+            //can solve the dup with the map here by doing a better handling
+            return;
+        }
+        ItemStack item = ((ItemFrame) entity).getItem();
+        if (item.getType() == Material.FILLED_MAP) {
+            //if the map exist we canceled the event
+            if (MapManager.getMap(item) != null) {
+                event.setCancelled(true);
+            }
+        }
+    }
+
+    @EventHandler
+    public void onEntityDamageByBlockEvent(EntityDamageByBlockEvent event) {
+        //Negate damage done to IoM maps by some blocks
+
+        Entity entity = event.getEntity();
+        if (!(entity instanceof ItemFrame)) {
+            return;
+        }
+        ItemStack item = ((ItemFrame) entity).getItem();
+        if (item.getType() == Material.FILLED_MAP) {
+            //if the map exist we canceled the event
+            if (MapManager.getMap(item) != null) {
+                switch (event.getCause()) {
+                    case MAGIC:
+                    case ENTITY_EXPLOSION:
+                    case FIRE_TICK:
+                    case LIGHTNING:
+                    case CRAMMING:
+                    case WITHER:
+                    case SUFFOCATION:
+                    case DROWNING:
+                    case BLOCK_EXPLOSION:
+                        event.setCancelled(true);
+                        break;
+                    default:
+                }
```

Comment:
Just squash this to `event.setCancelled(true);`

Reply:
No I don't want to cancel each causes.